lib/mongo/server/app_metadata/environment.rb

```diff
@@ -192,6 +211,32 @@ def detect_environment
           names.first
         end
 
+        # Looks for the presence of a container. Currently can detect
+        # Docker (by the existence of a Dockerfile in the working
+        # directory) and Kubernetes (by the existence of the KUBERNETES_SERVICE_HOST
+        # environment variable).
+        def detect_container
+          runtime = docker_present? && 'docker'
+          orchestrator = kubernetes_present? && 'kubernetes'
+
+          if runtime || orchestrator
+            fields[:container] = {}
+            fields[:container][:runtime] = runtime if runtime
+            fields[:container][:orchestrator] = orchestrator if orchestrator
+          end
+        end
+
+        # Checks for the existence of a Dockerfile in the working directory.
+        def docker_present?
+          File.exist?('Dockerfile')
```

Comment:
I believe that we should check for the presence of `.dockerenv` here - https://github.com/mongodb/specifications/pull/1454/files#diff-0ecfac0976ff0bfe5b91b5bd52bc5a79dd1125434505ff22331c7cadf1b2f25eR379